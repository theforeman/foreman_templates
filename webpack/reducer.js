import { combineReducers } from 'redux';

import syncSettings from './components/NewTemplateSync/NewTemplateSyncReducer';
import syncResult from './components/TemplateSyncResult/TemplateSyncResultReducer';

export default combineReducers({ syncSettings, syncResult });
