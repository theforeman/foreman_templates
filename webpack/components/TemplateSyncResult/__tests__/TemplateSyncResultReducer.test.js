import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';

import reducer, { initialState } from '../TemplateSyncResultReducer';
import { importTemplates } from '../__fixtures__/templateSyncResult.fixtures';

import {
  TEMPLATESYNC_FORM_SUBMITTED,
  SYNC_RESULT_PAGINATION_CHANGE,
} from '../../../consts';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },
  'should set new pagination': {
    state: initialState,
    action: {
      type: SYNC_RESULT_PAGINATION_CHANGE,
      payload: {
        pagination: {
          page: 2,
          perPage: 5,
        },
      },
    },
  },
  'should update state when form submitted': {
    state: initialState,
    action: {
      type: TEMPLATESYNC_FORM_SUBMITTED,
      payload: {
        data: {
          templates: importTemplates,
          repo: 'https://github.com/theforeman/community-templates.git',
          branch: 'master',
          result_action: 'import',
        },
      },
    },
  },
};

describe('TemplateSyncResultReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
