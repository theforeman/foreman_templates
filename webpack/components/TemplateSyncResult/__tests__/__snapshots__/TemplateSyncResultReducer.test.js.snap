// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateSyncResultReducer should return initial state 1`] = `
Object {
  "pagination": Object {
    "page": 1,
    "perPage": 20,
  },
  "resultAction": "",
  "templates": Array [],
}
`;

exports[`TemplateSyncResultReducer should set new pagination 1`] = `
Object {
  "pagination": Object {
    "page": 2,
    "perPage": 5,
  },
  "resultAction": "",
  "templates": Array [],
}
`;

exports[`TemplateSyncResultReducer should update state when form submitted 1`] = `
Object {
  "branch": "master",
  "pagination": Object {
    "page": 1,
    "perPage": 20,
  },
  "repo": "https://github.com/theforeman/community-templates.git",
  "resultAction": "import",
  "templates": Array [
    Object {
      "additionalErrors": "No \\"name\\" found in metadata",
      "errors": null,
      "name": null,
      "templateFile": "random_template.erb",
    },
    Object {
      "additionalErrors": null,
      "className": "ProvisioningTemplate",
      "errors": Object {
        "base": Array [
          "This template is locked",
        ],
      },
      "humanizedClassName": "Provisioning Template",
      "kind": "Provision",
      "locked": true,
      "name": "EPEL",
      "snippet": true,
      "templateFile": "epel.erb",
    },
    Object {
      "additionalErrors": null,
      "className": "Ptable",
      "errors": Object {},
      "humanizedClassName": "Ptable",
      "locked": false,
      "name": "CoreOS default",
      "snippet": false,
      "templateFile": "coreos_default.erb",
    },
    Object {
      "additionalErrors": "No \\"model\\" found in metadata",
      "errors": null,
      "name": null,
      "templateFile": "no_model.erb",
    },
    Object {
      "additionalErrors": null,
      "className": "ProvisioningTemplate",
      "errors": Object {
        "name": Array [
          "has already been taken",
        ],
      },
      "humanizedClassName": "Provisioning Template",
      "locked": true,
      "name": "Kickstart fake",
      "snippet": true,
      "templateFile": "kickstart_fake.erb",
    },
  ],
}
`;
